packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,241 @@
+   */
+  createRenderRoot() {
+    return this;
+  }
+
+  /**
+   * Adds event listeners.
+   */
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+      this.updateText
+    );
+  }
+
+  /**
+   * Removes event listeners.
+   */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+      this.updateText
+    );
+  }
+
+  /**
+   * Dispatches a search request event with the `value` property or string passed in.
+   * Invoked in response to user interactions: `enter` key or click on search button.
+   */
+  emitSearchEvent(query?: String) {
+    const term = query && typeof query === 'string' ? query : this.value;
+    const searchboxRequestEvent = new CustomEvent(
+      SEARCHBOX_EVENT.SEARCH_REQUEST,
+      {
+        detail: term,
+        bubbles: true
+      }
+    );
+    window.dispatchEvent(searchboxRequestEvent);
+  }
+
+  /**
+   * Dispatches an autocomplete request event.
+   * Invoked in response to user input: typed search term is greater than 3 characters.
+   *
+   * @param letters A string of letters inputed into input box.
+   */
+  emitAutocompleteRequestEvent(letters: String) {
+    const autocompleteRequestEvent = new CustomEvent(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_REQUEST,
+      { detail: letters, bubbles: true }
+    );
+    window.dispatchEvent(autocompleteRequestEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the input box has been cleared.
+   * Invoked in response to user input: clear button, remove all letters
+   * using keypresses.
+   */
+  emitSearchBoxClearedEvent() {
+    const searchboxClearedEvent = new CustomEvent(
+      SEARCHBOX_EVENT.SEARCHBOX_CLEARED
+    );
+    window.dispatchEvent(searchboxClearedEvent);
+  }
+
+  /**
+   * Updates the contents of the search input box with the payload of the given
+   * event.
+   * Invoked in response to an autocomplete hover event.
+   *
+   * @param e The event object.
+   */
+  updateText(e: CustomEvent) {
+    this.value = e.detail;
+    let el = this.getInputElement();
+    el.value = e.detail;
+  }
+
+  /**
+   * Returns the searchbox input element
+   */
+  getInputElement() {
+    return <HTMLInputElement>this.querySelector('#searchInput');
+  }
+
+  /**
+   * Receives user input from searchbox and calls appropriate function based
+   * on the input value.
+   *
+   * Invoked in response to a keyup.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleKeyup(e: KeyboardEvent) {
+    if (e.keyCode === 8) {
+      if (this.value.length === 1) {
+        this.emitSearchBoxClearedEvent();
+      }
+      this.value = this.value.slice(0, this.value.length - 1);
+    } else if (e.keyCode === 13 && this.value.length > 0) {
+      this.emitSearchEvent(this.value);
+    } else {
+      this.updatevalue((<HTMLInputElement>e.target).value);
+      if (this.value.length > 3) {
```

Comment:
As I mentioned before, I think that the number of characters that an autocomplete request is triggered on should be configurable, so I think that this will have to change.